cmd/vcluster/cmd/values/v1/base_values.go

```diff
@@ -0,0 +1,344 @@
+	} `json:"role,omitempty"`
+}
+
+type PDBValues struct {
+	Enabled bool `json:"enabled,omitempty"`
+	policyv1.PodDisruptionBudgetSpec
+}
+
+type ServiceValues struct {
+	Type                     corev1.ServiceType                  `json:"type,omitempty"`
+	ExternalIPs              []string                            `json:"externalIPs,omitempty"`
+	ExternalTrafficPolicy    corev1.ServiceExternalTrafficPolicy `json:"externalTrafficPolicy,omitempty"`
+	LoadBalancerIP           string                              `json:"loadBalancerIP,omitempty"`
+	LoadBalancerSourceRanges []string                            `json:"loadBalancerSourceRanges,omitempty"`
+	LoadBalancerClass        string                              `json:"loadBalancerClass,omitempty"`
+}
+
+type IngressValues struct {
+	Enabled          bool                      `json:"enabled,omitempty"`
+	PathType         string                    `json:"pathType,omitempty"`
+	APIVersion       string                    `json:"apiVersion,omitempty"`
+	IngressClassName string                    `json:"ingressClassName,omitempty"`
+	Host             string                    `json:"host,omitempty"`
+	Annotations      map[string]string         `json:"annotations,omitempty"`
+	TLS              []networkingv1.IngressTLS `json:"tls,omitempty"`
+}
+
+type CoreDNSValues struct {
+	Integrated     bool                        `json:"integrated,omitempty"`
+	Plugin         CoreDNSPluginValues         `json:"plugin,omitempty"`
+	Enabled        bool                        `json:"enabled,omitempty"`
+	Replicas       uint32                      `json:"replicas,omitempty"`
+	NodeSelector   corev1.NodeSelector         `json:"nodeSelector,omitempty"`
+	Image          string                      `json:"image,omitempty"`
+	Config         string                      `json:"config,omitempty"`
+	Service        CoreDNSServiceValues        `json:"service,omitempty"`
+	Resources      corev1.ResourceRequirements `json:"resources,omitempty"`
+	Manifests      string                      `json:"manifests,omitempty"`
+	PodAnnotations map[string]string           `json:"podAnnotations,omitempty"`
+	PodLabels      map[string]string           `json:"podLabels,omitempty"`
+}
+
+type CoreDNSPluginValues struct {
+	Enabled bool
+	Config  []DNSMappings
+}
+
+type DNSMappings struct {
+	Record    Record       `json:"record"`
+	Target    Target       `json:"target"`
+	AllowedOn []FilterSpec `json:"allowedOn"`
+	ExceptOn  []FilterSpec `json:"exceptOn"`
+}
+
+type Record struct {
+	RecordType RecordType
+	FQDN       *string `json:"fqdn"`
+	Service    *string `json:"service"`
+	Namespace  *string `json:"namespace"`
+}
+
+type RecordType string
+type TargetMode string
+
+type Target struct {
+	Mode      TargetMode `json:"mode"`
+	Vcluster  *string    `json:"vcluster"`
+	URL       *string    `json:"url"`
+	Service   *string    `json:"service"`
+	Namespace *string    `json:"namespace"`
```

Comment:
Is omitempty missing here on purpose?

Reply:
Actually all these record related values are basically coming from pro code, and there we do have things like enum defined for these types. I just kept them same to have an easier parity amount these values across the repos